### ISSUE TITLE
Accessibility feature not yet working properly on macos (VoiceOver)

### ISSUE THREAD

**Issue by sebkur** [JetBrains/compose-jb#1772]
I'm trying out the accessibility feature on macos. I've read [here](https://blog.jetbrains.com/kotlin/2021/10/compose-multiplatform-goes-beta/#accessibility) that they should be somewhat working there. I went to system preferences, accessibility and enabled VoiceOver. Usually when selecting UI elements such as entering any text input etc. this makes a bounding box appear around the element and generates some voice output on the speaker about that element. What I experience with compose is that the first UI element on screen gets read out aloud, but it is not possible to get VoiceOver to read out anything about other UI elements.

Here's a screenshot of how VoiceOver highlights the top left widget in the widgets gallery:

**Comment by sebkur** [comment]
<img width="798" alt="Screenshot 2022-01-31 at 18 28 28" src="https://user-images.githubusercontent.com/1633488/151844371-02d55b85-b419-4f16-94df-76417f2e5504.png">

**Comment by sebkur** [comment]
I can get VoiceOver to tell me stuff about other elements, but only by modifying the source code and making some different widget the first one on the screen. For example when I create a text input as first element, VoiceOver tells me something about that text field.

I'm wondering: what did you test accessibility with when implementing it for Compose? Something else than VoiceOver that ships with macOS?

**Comment by sebkur** [comment]
@elioqoshi

**Comment by VeselovAlex** [comment]
Hi, @sebkur 
Try to navigate with Control+Option+Up/Down Arrow, and follow the VoiceOver hints, e.g.

https://user-images.githubusercontent.com/3617329/152356382-7b40396e-a70f-44a5-9249-e55f8dbe2fd7.mov

Also you can use Accessibility Inspector for debugging, it is shipped with Xcode

**Comment by elioqoshi** [comment]
@VeselovAlex Thanks for the reply! This indeed works, although the focus order is different from what happens when navigating via Tab/Tab+Shift just like anyone would do on a website. How come the classic navigation with tab control is not possible?

**Comment by VeselovAlex** [comment]
Hi, @elioqoshi ! Seems that tab navigation works properly in master branch

https://user-images.githubusercontent.com/3617329/152534937-ca4051bb-22cc-4eef-95f7-e8768ae6e078.mov

**Comment by elioqoshi** [comment]
@VeselovAlex Yeah I got that. What I am saying is that VoiceOver doesn't work when navigating via Tab.

**Comment by sebkur** [comment]
yes, or for moving focus to a specific widget such as a `TextField` for example by clicking it. I think that should also make the accessibility software like VoiceOver pick up that this widget now has the focus and make it read out some text about that. I think probably the accessibility feature and the focus management should be linked somehow?

**Comment by VeselovAlex** [comment]
Tab navigation should work in 1.1.0-rc1

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-1772">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
